src/transport_ofi.c

```diff
@@ -1354,32 +1347,39 @@ static int shmem_transport_ofi_ctx_init(shmem_transport_ctx_t *ctx, int id)
         shmem_transport_ofi_is_private(ctx->options)) {
             ctx->tid = shmem_transport_ofi_gettid();
     }
-    shmem_transport_ofi_stx_allocate(ctx);
 
-    ret = bind_enable_cntr_ep_resources(ctx);
-    OFI_CHECK_RETURN_MSG(ret, "context bind/enable CNTR endpoint failed (%s)\n", fi_strerror(errno));
+    /* STX allocation deferred to shmem_transport_startup for default context */
+    if (ctx->id != SHMEM_TRANSPORT_CTX_DEFAULT_ID) {
```

Comment:
Can we delay creating the default context to startup instead of adding this special case?